apiserver/common/modelmanagerinterface.go

```diff
@@ -101,13 +101,14 @@ var _ ModelManagerBackend = (*modelManagerStateShim)(nil)
 
 type modelManagerStateShim struct {
 	*state.State
-	pool *state.StatePool
+	model *state.Model
+	pool  *state.StatePool
 }
 
 // NewModelManagerBackend returns a modelManagerStateShim wrapping the passed
 // state, which implements ModelManagerBackend.
-func NewModelManagerBackend(st *state.State, pool *state.StatePool) ModelManagerBackend {
-	return modelManagerStateShim{st, pool}
+func NewModelManagerBackend(st *state.State, m *state.Model, pool *state.StatePool) ModelManagerBackend {
+	return modelManagerStateShim{st, m, pool}
```

Comment:
We could avoid the churn in all these constructors which take both a state and model by:
- change now to take just model arg
- add a State() api to Model (just temporary)
- use the State() api to get state inside ModelManagerBackend

then when state is no longer required, we can drop the API off Model